src/main/java/reactor/netty/http/client/HttpClient.java

```diff
@@ -657,7 +657,22 @@ public final HttpClient doAfterResponse(BiConsumer<? super HttpClientResponse, ?
 	 */
 	public final HttpClient doAfterResponseSuccess(BiConsumer<? super HttpClientResponse, ? super Connection> doAfterResponseSuccess) {
 		Objects.requireNonNull(doAfterResponseSuccess, "doAfterResponseSuccess");
-		return new HttpClientDoOn(this, null, null, null, null, doAfterResponseSuccess);
+		return new HttpClientDoOn(this, null, null, null, null, doAfterResponseSuccess, null);
+	}
+
+	/**
+	 * Setup a callback called when {@link HttpClientResponse} failed with
+	 * {@link reactor.netty.channel.AbortedException#isConnectionReset(Throwable)}
+	 * and {@link HttpClientState#RETRY_ENABLED} has been emitted.
+	 *
+	 * @param doOnRetryEnabled a callback called when {@link HttpClientResponse} failed with
+	 * {@link reactor.netty.channel.AbortedException#isConnectionReset(Throwable)}
+	 *
+	 * @return a new {@link HttpClient}
+	 */
+	public final HttpClient doOnRetryEnabled(BiConsumer<? super HttpClientResponse, ? super Connection> doOnRetryEnabled) {
```

Comment:
I suppose this is intended to allow tagging of the error that resulted in repeating the other events again? Or when this is called, it is not going to call something else maybe you can look at this state diagram and comment expectations here https://github.com/square/okhttp/blob/master/docs/events.md#events-with-retries-and-follow-ups

Reply:
hard to elaborate my question without a test.

I suppose I mean.. should we expect in retry this?

doOnRequest
doOnResponseError
doOnRetry
doOnRequest
doOnResponse

or

doOnRequest
doOnRetry
doOnRequest
doOnResponse

or something else... if the former I agree we can kill the doOnRetry as long as we can see the doOnResponseError

Reply:
My idea is to have this flow

```
doOnRequest
doOnRetry
doOnRequest
doOnResponse
```

Reply:
Retry is not performed for any exception so using this callback retry can be distinguish from error

Reply:
technically it is an exception :P just a matter if the callback sees it or not. Is the response object useful here? Would it not be better to provide the underlying throwable and/or replace with doOnError? Maybe I'm reading too much into the javadoc, but it seems like this is literally in response to a Throwable.